x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -166,12 +166,23 @@ func createCloudWatchEvents(getMetricDataResults []cloudwatch.MetricDataResult,
 
 	// AWS EC2 Metrics
 	mapOfMetricSetFieldResults := make(map[string]interface{})
-	for _, output := range getMetricDataResults {
-		if len(output.Values) == 0 {
-			continue
+
+	// Find a timestamp for all metrics in output
+	timestamp := aws.FindTimestamp(getMetricDataResults)
+	if !timestamp.IsZero() {
+		for _, output := range getMetricDataResults {
+			if len(output.Values) == 0 {
+				continue
+			}
+			exists, timestampIdx := aws.CheckTimestampInArray(timestamp, output.Timestamps)
+			if exists {
+				labels := strings.Split(*output.Label, " ")
+				// check timestamp to make sure metrics come from the same timestamp
+				if len(labels) == 2 && len(output.Values) > timestampIdx {
```

Comment:
In what cases is labels not 2 here?

Reply:
There is no case this should happen. Because labels are constructed here https://github.com/elastic/beats/blob/af2312041f1821195b4f78fa0ad1fde13effa0b3/x-pack/metricbeat/module/aws/ec2/ec2.go#L266 I can delete this check here.